tests/interface/TWAnyAddressTests.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "TWTestUtilities.h"
+
+#include "HexCoding.h"
+#include <TrustWalletCore/TWAnyAddress.h>
+#include <TrustWalletCore/TWCoinType.h>
+
+#include <gtest/gtest.h>
+
+using namespace TW;
+
+TEST(AnyAddress, InvalidString) {
+    auto string = STRING("0x4E5B2e1dc63F6b91cb6Cd759936495434C7e972F");
+    auto btcAddress = TWAnyAddressCreateWithString(string.get(), TWCoinTypeBitcoin);
+    auto ethAaddress = WRAP(TWAnyAddress, TWAnyAddressCreateWithString(string.get(), TWCoinTypeEthereum));
+
+    EXPECT_EQ(btcAddress, nullptr);
+    EXPECT_NE(ethAaddress.get(), nullptr);
+}
+
+TEST(AnyAddress, Data) {
+    // ethereum
+    {
+        auto string = STRING("0x4E5B2e1dc63F6b91cb6Cd759936495434C7e972F");
+        auto addr = WRAP(TWAnyAddress, TWAnyAddressCreateWithString(string.get(), TWCoinTypeEthereum));
```

Comment:
maybe do for look and put switch case? to make sure you always cover

Reply:
this data field is optional, only expose this when rpc needs it